buildSrc/src/main/kotlin/Java9Modularity.kt

```diff
@@ -0,0 +1,108 @@
+                    tasks.getByName(compilation.compileAllTaskName).dependsOn(compileModuleInfoTask)
+
+                    val checkModuleInfoTask = registerCheckModuleInfoTask(checkModuleInfoTaskName, modulePath, moduleInfoClassFile)
+                    checkModuleInfoTask.configure { dependsOn(compilation.compileAllTaskName) }
+                    tasks.getByName("check").dependsOn(checkModuleInfoTask)
+                }
+            }
+
+            if (multiRelease) {
+                tasks.getByName<Jar>(target.artifactsTaskName) {
+                    rename("module-info.class", "META-INF/versions/9/module-info.class")
+                    manifest {
+                        attributes("Multi-Release" to true)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun Project.registerCompileModuleInfoTask(taskName: String, modulePath: FileCollection, destinationDir: DirectoryProperty, moduleInfoSourceFile: File) =
+        tasks.register(taskName, JavaCompile::class) {
+            dependsOn(modulePath)
+            source(moduleInfoSourceFile)
+            classpath = files()
+            destinationDirectory.set(destinationDir)
+            sourceCompatibility = JavaVersion.VERSION_1_9.toString()
+            targetCompatibility = JavaVersion.VERSION_1_9.toString()
+            doFirst {
+                options.compilerArgs = listOf(
+                    "--release", "9",
+                    "--module-path", modulePath.asPath,
+                    "-Xlint:-requires-transitive-automatic"
+                )
+            }
+        }
+
+    private fun Project.registerCheckModuleInfoTask(taskName: String, modulePath: FileCollection, moduleInfoClassFile: File) =
+        tasks.register(taskName) {
+            dependsOn(modulePath)
+            doLast {
+                val jdeps = ToolProvider.findFirst("jdeps").orElseThrow { IllegalStateException("Tool 'jdeps' is not available") }
+                val moduleDescriptor = moduleInfoClassFile.inputStream().use { ModuleDescriptor.read(it) }
+                val moduleName = moduleDescriptor.name()
+                val expectedOutput = moduleDescriptor.toJdepsOutput(moduleInfoClassFile)
+
+                val outputCaptureStream = ByteArrayOutputStream()
+                val printStream = PrintStream(outputCaptureStream, true, Charsets.UTF_8)
+                jdeps.run(
+                    printStream, printStream,
+                    "--multi-release", "9",
+                    "--module-path", (modulePath + files(moduleInfoClassFile.parentFile)).asPath,
+                    "--check", moduleName
+                )
+                val actualOutput = outputCaptureStream.toString(Charsets.UTF_8).trim()
+
+                if (actualOutput != expectedOutput) {
+                    throw IllegalStateException("Module-info does not match!\n$actualOutput")
```

Comment:
```suggestion
                    throw IllegalStateException("Module-info requierements section does not match!\n$actualOutput")
```

Since we check only requirements, not exports, it's better to highlight this